pkg/entrypoint/server_test.go

```diff
@@ -0,0 +1,197 @@
+		}
+	})
+
+	It("should state of unknown command become failed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "unknowncommand",
+					Args:    []string{},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Failed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should state of not permission command become failed", func() {
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "./testresources/testscript.sh",
+					Args:    []string{},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Failed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+
+	It("should two sucessful command is executed in series", func() {
+
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "echo",
+					Args:    []string{"1"},
+				},
+				{
+					Name:    "test2",
+					Command: "sleep",
+					Args:    []string{"1"},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Completed"}, {Name: "test2", Status: "Running"}}},
+				{Jobs: []job{{Name: "test1", Status: "Completed"}, {Name: "test2", Status: "Completed"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+	It("should second command is pended when first command fails", func() {
+
+		testcase := testCase{
+			input: []Job{
+				{
+					Name:    "test1",
+					Command: "false",
+					Args:    []string{""},
+				},
+				{
+					Name:    "test2",
+					Command: "sleep",
+					Args:    []string{"1"},
+				}},
+			expected: []statusResponse{
+				{Jobs: []job{{Name: "test1", Status: "Failed"}, {Name: "test2", Status: "Pending"}}},
+			},
+		}
+
+		cancel := startRunner(apiAddr, testcase.input)
+		defer cancel()
+
+		for _, expected := range testcase.expected {
+			Eventually(gotStatus, 10, 0.5).Should(Equal(&expected))
+		}
+	})
+})
+
+func startRunner(listenAddr string, jobs []Job) context.CancelFunc {
+	ctx, cancel := context.WithCancel(context.Background())
+	runner := NewRunner(listenAddr, log, jobs)
+	go func() {
+		defer GinkgoRecover()
+		Expect(runner.Run(ctx)).To(Succeed())
+	}()
+	return cancel
+}
+
+func gotStatus(g Gomega) (*statusResponse, error) {
+	resp, err := http.Get(fmt.Sprintf("http://%v/status", apiAddr))
+	g.Expect(err).Should(Succeed())
+	res := &statusResponse{}
+	body, _ := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	json.Unmarshal(body, res)
+	return res, nil
+}
```

Comment:
This function returns `(*statusResponse, error)` but the second return value is not used.

It would be nice to omit the second return value, or to return err instead of expecting success in the function.